src/applications/mhv/medical-records/containers/App.jsx

```diff
@@ -105,26 +105,50 @@ const App = ({ children }) => {
           ],
         },
       ];
-      if (showNotes)
+      if (showNotes) {
+        const notesPath = '/summaries-and-notes';
+        const currentPath = location.pathname;
+
+        const activePath = currentPath.startsWith(notesPath)
+          ? currentPath
+          : notesPath;
+
         navPaths[0].subpaths.push({
-          path: '/summaries-and-notes',
+          path: activePath,
           label: 'Care summaries and notes',
           datatestid: 'care-summaries-and-notes-sidebar',
         });
-      if (showVaccines)
+      }
+      if (showVaccines) {
+        const vaccinesPath = '/vaccines';
+        const currentPath = location.pathname;
+
+        const activePath = currentPath.startsWith(vaccinesPath)
+          ? currentPath
+          : vaccinesPath;
+
         navPaths[0].subpaths.push({
-          path: '/vaccines',
+          path: activePath,
           label: 'Vaccines',
           datatestid: 'vaccines-sidebar',
         });
-      navPaths[0].subpaths.push({
-        path: '/allergies',
-        label: 'Allergies and reactions',
-        datatestid: 'allergies-sidebar',
-      });
+      }
+      if (showSideNav) {
+        const allergiesPath = '/allergies';
+        const currentPath = location.pathname;
+
+        const activePath = currentPath.startsWith(allergiesPath)
+          ? currentPath
+          : allergiesPath;
+        navPaths[0].subpaths.push({
+          path: activePath,
+          label: 'Allergies and reactions',
+          datatestid: 'allergies-sidebar',
+        });
+      }
```

Comment:
When in the details page, this code updates the link to point to the current record. The links in the sidenav should always point to the list page.
<img width="432" alt="image" src="https://github.com/department-of-veterans-affairs/vets-website/assets/87040148/64f898e8-4770-41c6-970f-d6b7c9069103">

Reply:
Fixed it.